lib/jekyll/utils.rb

```diff
@@ -139,7 +139,7 @@ def parse_date(input, msg = "Input could not be parsed.")
     # Returns true if the YAML front matter is present.
     # rubocop: disable PredicateName
     def has_yaml_header?(file)
-      !!(File.open(file, "rb", &:readline) =~ %r!\A---\s*\r?\n!)
+      !File.open(file, "rb", &:readline) !~ %r!\A---\s*\r?\n!
```

Comment:
You still have a `!` so this change does nothing but superficially confuse the user.  `!!` is straight forward and easy to spot from the get-go, where as `!` then later `!~` is confusing, without a significant performance improvement, this suites nothing.